tests/unit/domain/getCertificate.test.ts

```diff
@@ -0,0 +1,181 @@
+import { S3 } from 'aws-sdk';
+import CertificateNumberError from '../../../src/errors/CertificateNumberError';
+import MissingBucketNameError from '../../../src/errors/MissingBucketNameError';
+import NoBodyError from '../../../src/errors/NoBodyError';
+import VinError from '../../../src/errors/VinError';
+import CertificateDetails from '../../../src/interfaces/CertificateDetails';
+import getCertificate from '../../../src/domain/getCertificate';
+import IncorrectFileTypeError from '../../../src/errors/IncorrectFileTypeError';
+import MissingFolderNameError from '../../../src/errors/MissingFolderNameError';
+
```

Comment:
Do we also want to check our early exit conditions when we check the config?
There are a few instances where we throw even though we allow `undefined`